documentation/design-docs/diagnostics-client-library.md

```diff
@@ -0,0 +1,298 @@
+
+        // The default path that the diagnostics client library looks for the IPC socket (default: /tmp)
+        public static string DefaultIpcSocketPath { get; set; } 
+
+        /// <summary>
+        /// Get all the supported commands by this version of the diagnostics IPC.
+        /// </summary>
+        /// <returns>
+        /// IEnumerable of all supported commands Command Code.
+        /// </returns>
+        public static IEnumerable<int> GetAvailableCommands();
+
+        /// <summary>
+        /// Convert a command name in the right format (CommandSet:Command) - i.e. EventPipe.CollectTracing - to corresponding command code.
+        /// </summary>
+        /// <returns>
+        /// Corresponding command code for the given command name. This may throw UnknownCommandException.
+        /// </returns>
+        public static int GetCommandCodeFromCommandName(string commandName);
+
+        /// <summary>
+        /// Convert a command code to its string representation
+        /// </summary>
+        /// <returns>
+        /// Corresponding command name for the given command code. This may throw UnknownCommandException
+        /// </returns>
+        public static string GetCommandNameFromCommandCode(int commandCode);
+
+    }
+}
+```
+
+### EventPipeCommandHandler
+This is a CommandHandler class for sending EventPipe commands across the diagnostics IPC channel.
+```cs
+namespace Microsoft.Diagnostics.Client
+{
+    /// <summary>
+    /// A class for handling EventPipe commands.
+    /// </summary>
+    public class EventPipeCommandHandler
+    {
+        public ulong SessionId { get; }  /// A ulong representing the current session ID. Defaults to ulong.Max
+        public uint CircularBufferSizeMB { get; set; }   /// The maximum buffer size in MB. Defaults to 
+        public EventPipeSerializationFormat SerializationFormat { get; set; }  /// Serialization format. Defaults to NetTrace
+        
+        public EventPipeCommandHandler(int processId)
+
+        /// <summary>
+        /// Add a provider to the EventPipe command
+        /// </summary>
+        public void AddProvider(
+            string name,
+            ulong keywords=ulong.MaxValue,
+            EventLevel=EventLevel.Verbose,
+            string filterData=null)
+
+        /// <summary>
+        /// Add a path that this instances looks for the IPC socket.
+        /// </summary>
+        public void AddIpcSocketPath(string path)
+
+        /// <summary>
+        /// Start tracing the application via CollectTracing1 command.
+        /// </summary> 
+        /// <returns>
+        /// A Stream object to read
+        /// </returns> 
+        public Stream CollectTracing1(SessionConfiguration configuration)
+
+        /// <summary>
+        /// Start tracing the application via CollectTracing2 command.
+        /// </summary> 
+        /// <returns>
+        /// A Stream object to read
+        /// </returns> 
+        public Stream CollectTracing2()
```

Comment:
What's the deal with these names?

Reply:
They're the names given to the IPC protocol commands.. I'm not a fan of the names either but I think Noah suggested a good point so I can clean this up with better names.